materialfx/src/main/java/io/github/palexdev/materialfx/dialogs/MFXStageDialog.java

```diff
@@ -130,7 +131,42 @@ private void initialize() {
 			initDraggable();
 		});
 
-		addEventHandler(WindowEvent.WINDOW_SHOWN, event -> centerInOwner());
+                //
+                // When centerInownerNode is true, we want to show the dialog
+                // centered in the owner. However JavaFX calculates the actual
+                // width and height of the dialog only when it has to be shown,
+                // therefore there is no way to calculate its position relative
+                // to the owner in advance.
+                //
+                // The workaround is to detect when the dialog with and height
+                // change, so that as soon as there is a value the position of
+                // the dialog can be calculated and the dialog moved.
+                //
+                // Finally, we need anyway to center the dialog any time it is
+                // shown even if the dialog height and with have not changed.
+                //
+                onShowingProperty().set((t) -> {
+                    if (getWidth() == 0)  {
+                        return;
+                    }
+                    if (centerInOwnerNode.getValue()) {
+                        centerXInOwner(getWidth());
+                        centerYInOwner(getHeight());
+                    }
+                });
+
+		getContent().widthProperty().addListener((o, ov, nv) -> {
+                    if (centerInOwnerNode.getValue()) {
+                        centerXInOwner(nv.doubleValue());
+                    }
+                });
+
+                getContent().heightProperty().addListener((o, ov, nv) -> {
+                    if (centerInOwnerNode.getValue()) {
+                        centerYInOwner(nv.doubleValue());
+                    }
+                });
+
```

Comment:
I have some doubts here
1) When you need to handle something in development, you should avoid using methods like `setOnAction()`, `setOnShown()` and similar, because what happens when the user wants to set an action to perform in such cases? The handler set by the developer (us) is overridden and it won't work as intended anymore. `onShowingProperty().set()` should be replaced with `addEventFilter(WindowEvent.WINDOW_SHOWING, e -> {...})`
2) What happens if the stage dialog is created with no content? `NullPointerException` caused by `getContent()`
Also, what happens if the content is changes? Those listeners will be useless.
The correct way to do this is to make use of the recently introduced APIs on Observables
Example:
```java
contentProperty().flatMap(Region::widthProperty).addListener(...); // Same for height
```